stablehlo/reference/Element.cpp

```diff
@@ -268,6 +268,34 @@ Element Element::operator~() const {
       });
 }
 
+Element abs(const Element &el) {
+  Type type = el.getType();
+
+  if (isSupportedIntegerType(el.getType())) {
+    auto intEl = el.getIntegerValue();
+    return Element(type, intEl.abs());
+  }
+
+  if (isSupportedFloatType(type)) {
+    APFloat elVal = el.getFloatValue();
+    return Element(type, elVal.isNegative() ? -elVal : elVal);
+  }
+
+  if (isSupportedComplexType(type)) {
+    auto elVal = el.getComplexValue();
+    const llvm::fltSemantics &elSemantics = elVal.real().getSemantics();
+    auto resultVal = std::abs(std::complex<double>(
+        elVal.real().convertToDouble(), elVal.imag().convertToDouble()));
+    bool roundingErr;
+    APFloat result(resultVal);
+    result.convert(elSemantics, APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type.dyn_cast<ComplexType>().getElementType(), result);
```

Comment:
Line 284 is already checking that it is a ComplexType, so you can do a regular `cast` here.